### PR TITLE
feat(SDRS): add a new resource sdrs drill

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:
add a new resource sdrs drill
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

- This PR depends on #3117 and #3119
- Make sure the PR #3117 and #3119 has been merged before merge this PR.

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/sdrs' TESTARGS='-run TestAccDrill_basic'
...
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/sdrs -v -run TestAccDrill_basic -timeout 360m -parallel 4 
=== RUN   TestAccDrill_basic 
=== PAUSE TestAccDrill_basic
=== CONT  TestAccDrill_basic
--- PASS: TestAccDrill_basic (601.53s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sdrs      601.585s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/sdrs/resource_huaweicloud_sdrs_drill_test.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+}
+`, testDrill_base(name), name)
+}
+
+func testDrill_basic_update(name string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_sdrs_drill" "test" {
+  name     = "%[2]s_update"
```
Comment (iskey) [inline huaweicloud/services/acceptance/sdrs/resource_huaweicloud_sdrs_drill_test.go]: must keep alignment here.

File: docs/resources/sdrs_drill.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+## Example Usage
+
+```hcl
+variable "group_id" {}
+
+resource "huaweicloud_sdrs_drill" "test" {
+  name                 = "test-drill"
+  group_id             = var.group_id
+}
```
Comment (chengxiangdong) [inline docs/resources/sdrs_drill.md]:
```suggestion
resource "huaweicloud_sdrs_drill" "test" {
  name     = "test-drill"
  group_id = var.group_id
}
```

File: docs/resources/sdrs_drill.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
+---
+subcategory: "Storage Disaster Recovery Service (SDRS)"
+---
+
+# huaweicloud_sdrs_drill
+
+Manages an SDRS DR drill resource within HuaweiCloud.
+
+-> **NOTE:** This resource requires the protection group has at least one protected instance or replication pair.
```
Comment (chengxiangdong) [inline docs/resources/sdrs_drill.md]:
```suggestion
-> This resource requires the protection group has at least one protected instance or replication pair.
```

File: huaweicloud/services/acceptance/sdrs/resource_huaweicloud_sdrs_drill_test.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+  name = "%[2]s_drill_vpc"
+  cidr = "192.168.0.0/16"
+}
+
+resource "huaweicloud_vpc_subnet" "drill_vpc_subnet" {
+  name       = "%[2]s_drill_vpc_subnet"
+  vpc_id     = huaweicloud_vpc.drill_vpc.id
+  cidr       = "192.168.0.0/24"
+  gateway_ip = "192.168.0.2"
+}
```
Comment (chengxiangdong) [inline huaweicloud/services/acceptance/sdrs/resource_huaweicloud_sdrs_drill_test.go:84]: `common.TestBaseNetwork` includes `huaweicloud_vpc` and `huaweicloud_vpc_subnet` resources , which do not need to be created again.

File: huaweicloud/services/sdrs/resource_huaweicloud_sdrs_drill.go [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+			}
+		}
+		return diag.FromErr(err)
+	}
+
+	deleteTimeoutSec := int(d.Timeout(schema.TimeoutDelete).Seconds())
+	if err := drill.WaitForJobSuccess(client, deleteTimeoutSec, n.JobID); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
```
Comment (chengxiangdong) [inline huaweicloud/services/sdrs/resource_huaweicloud_sdrs_drill.go:171]:
```suggestion
	d.SetId("")
	return nil
```

**Review by chengxiangdong — COMMENTED** [review]
Nice job overall! 
Just a few nits: inconsistent indentation in a few places and do not forget to clean up the ID after deletion.

File: huaweicloud/services/acceptance/sdrs/resource_huaweicloud_sdrs_drill_test.go (hunk 4 above)
Comment (deer-hang) [reply huaweicloud/services/acceptance/sdrs/resource_huaweicloud_sdrs_drill_test.go:84]: The `huaweicloud_vpc` and `huaweicloud_vpc_subnet` of `common.TestBaseNetwork` used for protected instance.
SDRS drill need a new VPC.

File: huaweicloud/services/sdrs/resource_huaweicloud_sdrs_drill.go (hunk 5 above)
Comment (deer-hang) [reply huaweicloud/services/sdrs/resource_huaweicloud_sdrs_drill.go:171]: The deletion method will automatically set the ID to empty internally.

File: huaweicloud/services/acceptance/sdrs/resource_huaweicloud_sdrs_drill_test.go (hunk 4 above)
Comment (chengxiangdong) [reply huaweicloud/services/acceptance/sdrs/resource_huaweicloud_sdrs_drill_test.go:84]: ok, I get it

File: huaweicloud/services/sdrs/resource_huaweicloud_sdrs_drill.go (hunk 5 above)
Comment (chengxiangdong) [reply huaweicloud/services/sdrs/resource_huaweicloud_sdrs_drill.go:171]: ok, I get it

File: huaweicloud/services/acceptance/sdrs/resource_huaweicloud_sdrs_drill_test.go [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+resource "huaweicloud_sdrs_drill" "test" {
+  name         = "%[2]s"
+  group_id     = huaweicloud_sdrs_protection_group.test.id
+  drill_vpc_id = huaweicloud_vpc.drill_vpc.id
+
+  depends_on = [
+    huaweicloud_sdrs_protected_instance.test,
+  ]
+}
+`, testDrill_base(name), name)
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/sdrs/resource_huaweicloud_sdrs_drill_test.go:102]: can we use `testProtectedInstance_basic` function?

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/sdrs' TESTARGS='-run TestAccDrill_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/sdrs -v -run TestAccDrill_basic -timeout 360m -parallel 4 
=== RUN   TestAccDrill_basic 
=== PAUSE TestAccDrill_basic
=== CONT  TestAccDrill_basic
--- PASS: TestAccDrill_basic (508.95s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sdrs      509.007s
```

**Pull request merged**